docs/php-api.md

```diff
@@ -0,0 +1,734 @@
+
+### Function Signature
+
+```php
+wp_unregister_ability_category( string $slug ) ?\WP_Ability_Category
+```
+
+**Parameters:**
+- `$slug` (`string`): The slug of the registered category.
+
+**Return:** (`?\WP_Ability_Category`) The unregistered category instance on success, `null` on failure.
+
+## Fetch a Category
+
+Retrieve a specific category by slug.
+
+### Function Signature
+
+```php
+wp_get_ability_category( string $slug ) ?\WP_Ability_Category
+```
+
+**Parameters:**
+- `$slug` (`string`): The slug of the registered category.
+
+**Return:** (`?\WP_Ability_Category`) The category instance on success, `null` on failure.
+
+## Fetch all Categories
+
+Get all registered categories as an associative array keyed by slug.
+
+### Function Signature
+
+```php
+wp_get_ability_categories() array
+```
+
+**Return:** (`array`) An associative array of all registered categories, keyed by slug. Each value is an instance of `WP_Ability_Category`.
+
+## Registering Abilities (`wp_register_ability`)
+
+The primary way to add functionality to the Abilities API is by using the `wp_register_ability()` function, typically hooked into the `abilities_api_init` action.
+
+### Function Signature
+
+```php
+wp_register_ability( string $id, array $args ): ?\WP_Ability
+```
+
+- `$id` (`string`): A unique identifier for the ability.
+- `$args` (`array`): An array of arguments defining the ability configuration.
+- **Return:** (`?\WP_Ability`) An instance of the registered ability if it was successfully registered, `null` on failure (e.g., invalid arguments, duplicate ID).
+
+### Parameters Explained
+
+The `$args` array accepts the following keys:
+
+- `label` (`string`, **Required**): A human-readable name for the ability. Used for display purposes. Should be translatable.
+- `description` (`string`, **Required**): A detailed description of what the ability does, its purpose, and its parameters or return values. This is crucial for AI agents to understand how and when to use the ability. Should be translatable.
+- `category` (`string`, **Required**): The slug of the category this ability belongs to. The category must be registered before registering the ability using `wp_register_ability_category()`. Categories help organize and filter abilities by their purpose. See [Registering Categories](7.registering-categories.md) for details.
+- `input_schema` (`array`, **Required**): A JSON Schema definition describing the expected input parameters for the ability's execute callback. Used for validation and documentation.
+- `output_schema` (`array`, **Required**): A JSON Schema definition describing the expected format of the data returned by the ability. Used for validation and documentation.
+- `execute_callback` (`callable`, **Required**): The PHP function or method to execute when this ability is called.
+  - The callback receives one optional argument: it can have any type as defined in the input schema (e.g., `array`, `object`, `string`, etc.).
```

Comment:
It’s worth clarifying that that the param is optional and defaults to null only when no schema provided. The same applies to permission check callback.